schemas/bedrock-edv-storage.js

```diff
@@ -250,7 +250,126 @@ const query = {
   }
 };
 
-module.exports.config = () => edvConfig;
-module.exports.chunk = () => edvDocumentChunk;
-module.exports.document = () => edvDocument;
-module.exports.query = () => query;
+const getEdvsQuery = {
+  title: 'edv query',
+  type: 'object',
+  required: ['controller', 'referenceId'],
+  additionalProperties: false,
+  properties: {
+    controller: {
+      type: 'string'
+    },
+    referenceId: {
+      type: 'string'
+    }
+  }
+};
+
+const getAuthorizationsQuery = {
+  title: 'authorization query',
+  type: 'object',
+  required: ['id'],
+  additionalProperties: false,
+  properties: {
+    id: {
+      type: 'string'
+    },
+  }
+};
+
+const zcap = {
+  title: 'zcap',
+  type: 'object',
+  additionalProperties: false,
+  properties: {
```

Comment:
I would think we need some required properties here -- or is that handled elsewhere?

Reply:
This same schema is used in validators in several models (bedrock-profile-htttp and bedrock-kms-http that I know of), and none of the tests written have additional properties. How do I find and test for what other properties may be out there?

Reply:
My apologies for missing "required" in the sentence. I have added `required` to the validation schema.

Reply:
I think `invoker` may not be required ... so long as either `controller` or `delegator` are present. I'm not sure how to pull that off with json-schema. @mattcollier, have we dealt with this sort of thing before anywhere else?

I also think `allowedAction` probably isn't strictly required. The others seem proper to me.

Reply:
One technique used to deal with this sort of thing is to chain schemas together using an [allOf](https://json-schema.org/understanding-json-schema/reference/combining.html#allof).  In this case, there would be two schemas.

The first schema includes an 'anyOf' that validates 'required' in the various allowed combinations.
Then, the second schema is existing more rigorous schema that checks all the data types, but does *not* apply `required`.

Some of this sort of thing is applied in the ledger stack.

Reply:
I updated the schema using anyOf and allOf.